neuraxle/hyperparams/space.py

```diff
@@ -226,6 +228,76 @@ class HyperparameterSamples(RecursiveDict):
     def __init__(self, *args, separator=None, **kwds):
         super().__init__(*args, separator=separator, **kwds)
 
+    def compress(self) -> 'CompressedHyperparameterSamples':
+        """Compresses the HyperparameterSamples representation."""
+        return CompressedHyperparameterSamples(self)
+
+
+@dataclass
+class CompressedHyperparameter:
+    """CompressedHyperParameter"""
+    step_name: str
+    hyperparams: dict
+    ancestor_steps: list
+
+
+class CompressedHyperparameterSamples:
+    """
+    Short-hand representation of `HyperparameterSamples`
+    """
+
+    def __init__(self, hps: HyperparameterSamples):
+        """Takes in `HyperparameterSamples` object and generates a compressed _seq."""
+        if not hps or not isinstance(hps, HyperparameterSamples):
+            raise ValueError("pass a valid `HyperparameterSamples` object")
+        self._flat_hps: dict = hps.to_flat_dict()  # this will be used to reconstruct the hyperparameters from
+        # compressed format
+        self._separator: str = hps.separator
+        self._seq: List[dict] = self._convert()
+
+    def __str__(self) -> str:
+        """Prints the `CompressedHyperparameterSamples._seq`."""
+        return f"{self._seq}"
```

Comment:
It would be good to have an accurate usage example in the docstring with the before and after. In this docstring, or in the class-level docstring. I still wonder how it looks like. 

I suggest to do a unit test for the "before" and "after" of the conversion, and use the printed before and printed after to insert it in the documentation. An accurate example with a test would be desired.

Reply:
yes Even i have the same thing in mind. But before that, i just need to make sure the logic is okay. Bdw do you guys also do use doctests? (or) is it just an overhead to have it? @guillaume-chevalier

Reply:
@Rohith295 Nice call for the Doctests! I didn't know this existed. It would be interesting to set it up in the status checks on GitHub.